observer/obs_conf.go

```diff
@@ -121,7 +147,7 @@ func (c *ObsConf) MakeObserver() (*Observer, error) {
 	logger.Infof("Initializing boulder-observer daemon")
 	logger.Debugf("Using config: %+v", c)
 
-	monitors, errs, err := c.makeMonitors()
+	monitors, errs, err := c.makeMonitors(metrics)
 	if len(errs) != 0 {
 		logger.Errf("%d of %d monitors failed validation", len(errs), len(c.MonConfs))
```

Comment:
I noticed after fixing a panic caused by using an undefined prober type that this error message was showing the wrong number. Turns out that by continuing on to construct a monitor after getting an error from `probers.GetConfigurer()` we a broke an assumption that `len(errs)` equals the number of failed monitors. Since the logic in `makeMonitors()` can fail and return an error in two different places now, this assumption doesn't hold when the error is "this prober type doesn't exist". I can see a few different ways to address this problem, but I'd like to know what you recommend. My recommendation is to change these length checks here and at the end of `makeMonitors()` to check the length of the returned array of monitors instead. `len(c.MonConfs) - len(createdMonitors)` should give the number of failed monitors

Reply:
I tend to agree, and this is pretty much what I was doing previously. If I understood correctly though, @beautifulentropy [suggested against this](https://github.com/letsencrypt/boulder/pull/6290#discussion_r944878561).

> For this block, we want to perform all of our metric construction and registration at the top. This keeps the new logic completely separate from operations we always want to do (e.g. re-assigning `entry`, constructing the `monitor`, and incrementing `countMonitors` `false`/`true`.

Thus why I'm not doing it that way in the current iteration.

Reply:
I see what you're getting at, but you can do all the metric construction and registration first, and still bail out of the whole thing if it fails.

Reply:
Done!